### PR TITLE
[MNT] Removed the test case for checking next page's url

### LINKED ISSUE #540: Rework `next-page` component test
Currently, the last test case in the `next-page.cy.js` intercepts a `GET` request to assert over the url when the `next-page` button is clicked:
https://github.com/neurobagel/annotation_tool/blob/cd4a0a4b911a30c66954e1b985b3315939b095b1/cypress/component/next-page.cy.js#L144-L174

However, since `next-page.cy,js` is a component test, it's better to test the component itself (i.e., it's ability to display elements, fire events, ...) as opposed to how it interacts with the rest of the app. 
So this test case needs to be reworked to check whether the responsible mutation is called when the `next-page` button is clicked.

*This came to light when bumping cypress lead to the test failing in #536*

### PR DESCRIPTION
**Pull request by rmanaem**
Closes #540 

Changes proposed in this pull request:

- Removed the test case for checking next page's url, see [this comment](https://github.com/neurobagel/annotation_tool/issues/540#issuecomment-1719907990) for more details
- Includes changes from #536 
- Set the node version of cypress component test to 18 since the `annot-explanation.cy.js` was failing on default node version

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: cypress/component/annot-explanation.cy.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ describe("explanation", () => {
         cy.get(".btn").click();
 
         // Assert
-        cy.get(".card-body").should("be.visible");
+        cy.get(".card-body").should("be.visible",  { timeout: 8000 });
```
Comment (surchs) [inline cypress/component/annot-explanation.cy.js:64]: Why is this part of the PR?

File: package-lock.json
Comment (surchs) [inline package-lock.json:1]: seems unrelated

File: package.json
Comment (surchs) [inline package.json:1]: seems unrelated

**Review by surchs — COMMENTED** [review]
thanks @rmanaem! Agree with removing the test until https://github.com/neurobagel/annotation_tool/issues/545 is addressed. 

There seem to be some unrelated changes in the PR, could you check again that these make sense and are required. In general let's keep the changes in one PR focused to the issue so we can roll them back more easily if something breaks, and can reason better over what might be causing a new bug/breaking.

File: cypress/component/annot-explanation.cy.js (hunk 1 above)
Comment (rmanaem) [reply cypress/component/annot-explanation.cy.js:64]: This was inconsistent across different ci runs. For example, It failed [here](https://github.com/neurobagel/annotation_tool/actions/runs/6189686027/job/16805322365) when I was resolving the conflict.
It always passed locally so I checked the logs and it seemed to be timing out sometimes.

File: package.json
Comment (rmanaem) [reply package.json:1]: See #536

File: package-lock.json
Comment (rmanaem) [reply package-lock.json:1]: See #536

File: cypress/component/annot-explanation.cy.js (hunk 1 above)
Comment (surchs) [reply cypress/component/annot-explanation.cy.js:64]: ok, that's good to know. I'd say: add a short comment above this line to say that and then good to go

**Review by surchs — APPROVED** [review]
thanks, let's add a brief explainer comment for the custom timeout and then :cook:
